src/devices/machine/i8291a.h

```diff
@@ -0,0 +1,415 @@
+	static constexpr int REG_ADDRESS_STATUS_TPAS = 1 << 3;
+	static constexpr int REG_ADDRESS_STATUS_LPAS = 1 << 4;
+	static constexpr int REG_ADDRESS_STATUS_EOI = 1 << 5;
+	static constexpr int REG_ADDRESS_STATUS_LON = 1 << 6;
+	static constexpr int REG_ADDRESS_STATUS_TON = 1 << 7;
+
+	static constexpr int REG_AUXB_CPT_ENABLE = 1 << 0;
+	static constexpr int REG_AUXB_EOI_SPAS_ENABLE = 1 << 1;
+	static constexpr int REG_AUXB_HS_TRANSFER = 1 << 2;
+	static constexpr int REG_AUXB_INT_ACTIVE_LOW = 1 << 3;
+	static constexpr int REG_AUXB_RFD_HOLDOFF_GET_DEC = 1 << 4;
+
+	static constexpr int REG_AUXA_RFD_HOLDOFF_DATA = 1 << 0;
+	static constexpr int REG_AUXA_RFD_HOLDOFF_END = 1 << 1;
+	static constexpr int REG_AUXA_END_ON_EOS = 1 << 2;
+	static constexpr int REG_AUXA_EOI_ON_EOS = 1 << 3;
+	static constexpr int REG_AUXA_EOS_8BIT = 1 << 4;
+
+	// AUX CMDs
+	static constexpr int AUXCMD_IMMEDIATE_EXEC_PON = 0;
+	static constexpr int AUXCMD_CLEAR_PP = 1;
+	static constexpr int AUXCMD_CHIP_RESET = 2;
+	static constexpr int AUXCMD_FINISH_HANDSHAKE = 3;
+	static constexpr int AUXCMD_TRIGGER = 4;
+	static constexpr int AUXCMD_CLEAR_RTL = 5;
+	static constexpr int AUXCMD_SEND_EOI = 6;
+	static constexpr int AUXCMD_NON_VALID_SA = 7;
+	static constexpr int AUXCMD_PON = 8;
+	static constexpr int AUXCMD_SET_PP = 9;
+	static constexpr int AUXCMD_SET_RTL = 13;
+	static constexpr int AUXCMD_VALID_SA = 15;
+
+	// Interface commands
+	// TODO: stolen from tms9914, move to common header file. PHI also defines this
+
+	static constexpr uint8_t IFCMD_MASK       = 0x7f;  // Mask of valid bits in if. commands
+	static constexpr uint8_t IFCMD_ACG_MASK   = 0x70;  // Mask of ACG commands
+	static constexpr uint8_t IFCMD_ACG_VALUE  = 0x00;  // Value of ACG commands
+	static constexpr uint8_t IFCMD_UCG_MASK   = 0x70;  // Mask of UCG commands
+	static constexpr uint8_t IFCMD_UCG_VALUE  = 0x10;  // Value of UCG commands
+	static constexpr uint8_t IFCMD_GROUP_MASK = 0x60;  // Mask of group id
+	static constexpr uint8_t IFCMD_LAG_VALUE  = 0x20;  // Value of LAG commands
+	static constexpr uint8_t IFCMD_TAG_VALUE  = 0x40;  // Value of TAG commands
+	static constexpr uint8_t IFCMD_SCG_VALUE  = 0x60;  // Value of SCG commands
+	static constexpr uint8_t IFCMD_GTL        = 0x01;  // Go to local
+	static constexpr uint8_t IFCMD_SDC        = 0x04;  // Selected device clear
+	static constexpr uint8_t IFCMD_GET        = 0x08;  // Group execute trigger
+	static constexpr uint8_t IFCMD_TCT        = 0x09;  // Take control
+	static constexpr uint8_t IFCMD_LLO        = 0x11;  // Local lock-out
+	static constexpr uint8_t IFCMD_DCL        = 0x14;  // Device clear
+	static constexpr uint8_t IFCMD_SPE        = 0x18;  // Serial poll enable
+	static constexpr uint8_t IFCMD_SPD        = 0x19;  // Serial poll disable
+	static constexpr uint8_t IFCMD_UNL        = 0x3f;  // Unlisten
+	static constexpr uint8_t IFCMD_UNT        = 0x5f;  // Untalk
+
+	enum class SourceHandshake {
```

Comment:
Please be consistent and use `snake_case` for type names - we use `LlamaCase` for template parameters.